daemons/based/pacemaker-based.c

```diff
@@ -52,159 +55,190 @@ static bool startCib(const char *filename);
+    }
 
-    cib_writer = mainloop_add_trigger(G_PRIORITY_LOW, write_cib_contents, NULL);
+    if (args->version) {
+        g_strfreev(processed_args);
+        pcmk__free_arg_context(context);
 
-    while (1) {
-        flag = pcmk__next_cli_option(argc, argv, &index, NULL);
-        if (flag == -1)
-            break;
+        /* FIXME: When pacemaker-based is converted to use formatted output,
+         * this can go.
+         */
+        pcmk__cli_help('v', CRM_EX_OK);
+    }
 
-        switch (flag) {
-            case 'V':
-                crm_bump_log_level(argc, argv);
-                break;
-            case 's':
-                stand_alone = TRUE;
-                preserve_status = TRUE;
-                cib_writes_enabled = FALSE;
-
-                pwentry = getpwnam(CRM_DAEMON_USER);
-                CRM_CHECK(pwentry != NULL,
-                          crm_perror(LOG_ERR, "Invalid uid (%s) specified", CRM_DAEMON_USER);
-                          return CRM_EX_FATAL);
-
-                rc = setgid(pwentry->pw_gid);
-                if (rc < 0) {
-                    crm_perror(LOG_ERR, "Could not set group to %d", pwentry->pw_gid);
-                    return CRM_EX_FATAL;
-                }
+    mainloop_add_signal(SIGTERM, cib_shutdown);
+    mainloop_add_signal(SIGPIPE, cib_enable_writes);
 
-                rc = initgroups(CRM_DAEMON_USER, pwentry->pw_gid);
-                if (rc < 0) {
-                    crm_perror(LOG_ERR, "Could not setup groups for user %d", pwentry->pw_uid);
-                    return CRM_EX_FATAL;
-                }
+    cib_writer = mainloop_add_trigger(G_PRIORITY_LOW, write_cib_contents, NULL);
 
-                rc = setuid(pwentry->pw_uid);
-                if (rc < 0) {
-                    crm_perror(LOG_ERR, "Could not set user to %d", pwentry->pw_uid);
-                    return CRM_EX_FATAL;
-                }
-                break;
-            case '?':          /* Help message */
-                pcmk__cli_help(flag, CRM_EX_OK);
-                break;
-            case 'w':
-                cib_writes_enabled = TRUE;
-                break;
-            case 'r':
-                cib_root = optarg;
-                break;
-            case 'm':
-                cib_metadata();
-                return CRM_EX_OK;
-            default:
-                ++argerr;
-                break;
-        }
-    }
-    if (argc - optind == 1 && pcmk__str_eq("metadata", argv[optind], pcmk__str_casei)) {
+    if ((g_strv_length(processed_args) >= 2)
+        && pcmk__str_eq(processed_args[1], "metadata", pcmk__str_none)) {
         cib_metadata();
-        return CRM_EX_OK;
+        goto done;
     }
 
-    if (optind > argc) {
-        ++argerr;
-    }
+    crm_log_init(NULL, LOG_INFO, TRUE, FALSE, argc, argv, FALSE);
 
-    if (argerr) {
-        pcmk__cli_help('?', CRM_EX_USAGE);
+    for (int i = 0; i < args->verbosity; i++) {
+        crm_bump_log_level(argc, argv);
     }
 
```

Comment:
You can use pcmk__cli_init_logging instead of this block, too.  However since this is in a daemon, you should then move the crm_log_init call to afterwards (see daemons/schedulerd/pacemaker-scheduler.c).